ext/RMagick/rmilist.c

```diff
@@ -504,13 +505,23 @@ ImageList_montage(VALUE self)
 
     images = images_from_imagelist(self);
 
-    // If app specified a non-default composition operator, use it for all images.
-    if (montage->compose != UndefinedCompositeOp)
+    for (image = images; image; image = GetNextImageInList(image))
     {
-        Image *i;
-        for (i = images; i; i = GetNextImageInList(i))
+        if (montage->compose != UndefinedCompositeOp)
+        {
+            image->compose = montage->compose;
+        }
+        if (montage->change.background_color)
+        {
+            image->background_color = montage->info->background_color;
+        }
+        if (montage->change.border_color)
```

Comment:
I really wonder if we need these values to check if it was set by the user. Shouldn't we always set the `*_color` to the value of the montage info?

Reply:
I think we should apply the value only when user set. I don't know what correct usage in ImageMagick. But, if we don't use MontageInfo for color, we have to change the interface.

Reply:
But if the image background color is red and the `MontageInfo` has the default color (white/black?) should we not use that instead?

Reply:
we should set the color what user desired via Montage#border_color ....

Reply:
But why should we only do this when the user has changed the value? Shouldn't we always use the default value instead? If the users sees the color is black and decides not to change it because that is the value that they want to use but then we don't use it because the user did not explicitly set it?

Reply:
If image has user expected value, I think user doesn't need to change.

Reply:
This the call that is being made: `MontageImages(images, montage->info, exception);`. It is now possible that `montage->info->border_color` has a different value than `image->border_color`. I wonder if the `changed` checks should be removed to make sure they have the same value.

Reply:
I'm not sure I fully understand this, but I think I agree with @dlemstra. If the user calls the method, I think it should have *some* result, even if they haven't put in values, rather than silently doing nothing. If we do nothing, it could be confusing to the user why nothing is happening, but if we change the background color to something different than what they expect, they'll have a better idea of where to look. It also seems like it'll make for much cleaner code.